### PR TITLE
Security rest resource

### PR DESCRIPTION
**Pull request by morihladko**
I created a `guard` class decorator, which takes a list of policies to guard the execution of CRUD methods on given resource.

### CODE REVIEW

File: zsl/resource/guard.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+        setattr(cls, '_guard_policies', list(self.policies))
+
+        for method_name in self.resource_methods:
+            secure_name = _secure_name(method_name)
+
+            if hasattr(cls, method_name):
+                method = getattr(cls, method_name)
+
+                setattr(cls, method_name, self._wrap(method))
+                setattr(cls, secure_name, method)
```
Comment (babky-atteq) [inline zsl/resource/guard.py]: should not the secured method be here as well?

Comment (babky-atteq) [inline zsl/resource/guard.py]: I see the secure_name is the non-secured method (core of the functionality). this is something that got me.

File: tests/resource/json_server_model_resource_test.py [hunk 2]
```diff
@@ -26,6 +23,7 @@
     UserModel, UserTuple, create_resource_test_data, users, test_settings,
     get_non_existent_id, addresses
 )
+from tests.test_utils import parent_module, json_loads, HttpTestCase
 
 # py23
```
Comment (morihladko) [inline tests/resource/json_server_model_resource_test.py]: :P it's python 2/3 compatibility mark

Comment (babky-atteq) [inline tests/resource/json_server_model_resource_test.py]: ok

File: zsl/resource/guard.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+from functools import wraps
+
+from zsl.interface.resource import ResourceResult
+from zsl.service.service import TransactionalSupport, transactional
+
+
+class ResourcePolicy(object):
+    """Declarative policy class.
+
+    Every CRUD method has is corespondent *can_method__before* and
```
Comment (babky-atteq) [inline zsl/resource/guard.py]: ok

File: zsl/resource/guard.py (hunk 1 above)
Comment (babky-atteq) [inline zsl/resource/guard.py]: guarded_read

**Pull request merged**